README.md

```diff
@@ -50,6 +50,42 @@ have dependencies on other algebras which must be implemented.
     - Two promises are equivalent when they yield equivalent values.
     - Two functions are equivalent if they yield equivalent outputs for equivalent inputs.
 
+## Type signature notation
+
+The type signature notation used in this document is described
+[here](https://sanctuary.js.org/#types). In summary:
+
+1. _New types can be created via type constructors._
+    - Type constructors take one or more type arguments. For example, the type
+    "array of strings" is written as `Array String`, and the type constructor
+    `Array` can be defined as `Array :: Type -> Type`.
```

Comment:
I think this point should come last, or be removed. An explanation of the notation should start with the simplest possible example and build from there. Type constructors should not be introduced until the groundwork has been laid.

I suggest something like this:

```haskell
-- simple example
true :: Boolean
```

```haskell
-- show that a value can be a member of multiple types
42 :: Integer, Number
```

```haskell
-- introduce type constructors
["foo", "bar", "baz"] :: Array String
```

```haskell
-- introduce the (->) type constructor
Math.abs :: Number -> Number
```